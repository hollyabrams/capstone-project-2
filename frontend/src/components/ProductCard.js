import { Link } from "react-router-dom";
import { useShoppingCart, formatCurrencyString } from "use-shopping-cart";
import { toast } from "react-hot-toast";

export default function ProductCard({ product, index }) {
    const { addItem } = useShoppingCart()

    function onAddToCart(event) {
        event.preventDefault()
        const id = toast.loading("Adding 1 item...")
        addItem(product)
        toast.success(`${product.name} added`, { id })
    }

    return (
        <Link to={`/products/${product.id}`} className="border-2 rounded-md overflow-hidden">
            <div className="relative w-full h-64">
                <img
                    src={product.image}
                    alt={product.name}
                    style={{
                        objectFit: 'contain',
                        width: '100%',
                        height: '100%',
                    }}
                />
            </div>

            <div className="p-6 bg-white">
                <p className="font-semibold text-lg">{ product.name }</p>
                <div className="mt-4 flex items-center justify-between space-x-2">
                    <div>
                        <p className="text-gray-500">Price</p>
                        <p className="text-lg font-semibold">{formatCurrencyString({
                            currency: product.currency,
                            value: product.price,
                            })}
                        </p>
                    </div>
                    <button onClick={onAddToCart} className="border rounded-lg py-1 px-4 transition-colors duration-200 ease-in-out 
                    hover:bg-lime-500 hover:text-white">Add to Cart</button>
                </div>
            </div>
        </Link>
    );
}

