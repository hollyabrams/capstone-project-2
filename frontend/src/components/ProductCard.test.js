import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { useShoppingCart, formatCurrencyString } from "use-shopping-cart";
import { toast } from "react-hot-toast";
import ProductCard from './ProductCard';

jest.mock("use-shopping-cart", () => ({
  useShoppingCart: () => ({
    addItem: jest.fn()
  }),
  formatCurrencyString: jest.fn(({ value, currency }) => `${currency}${(value/100).toFixed(2)}`) // More accurate mock
}));

jest.mock("react-hot-toast");

const mockProduct = {
  id: 1,
  name: "Test Product",
  image: "test.jpg",
  price: 1000,
  currency: "$"
};

test('ProductCard matches snapshot', () => {
  const { asFragment } = render(
    <Router>
      <ProductCard product={mockProduct} index={0} />
    </Router>
  );
  
  expect(asFragment()).toMatchSnapshot();
});
