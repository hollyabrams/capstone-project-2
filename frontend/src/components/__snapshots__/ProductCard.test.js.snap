// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="border-2 rounded-md overflow-hidden"
    href="/products/1"
  >
    <div
      class="relative w-full h-64"
    >
      <img
        alt="Test Product"
        src="test.jpg"
        style="object-fit: contain; width: 100%; height: 100%;"
      />
    </div>
    <div
      class="p-6 bg-white"
    >
      <p
        class="font-semibold text-lg"
      >
        Test Product
      </p>
      <div
        class="mt-4 flex items-center justify-between space-x-2"
      >
        <div>
          <p
            class="text-gray-500"
          >
            Price
          </p>
          <p
            class="text-lg font-semibold"
          />
        </div>
        <button
          class="border rounded-lg py-1 px-4 transition-colors duration-200 ease-in-out  hover:bg-lime-500 hover:text-white"
        >
          Add to Cart
        </button>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
